src/plots/cartesian/layout_attributes.js

```diff
@@ -698,6 +698,26 @@ module.exports = {
             'In other cases the default is *hide past div*.'
         ].join(' ')
     },
+    ticklabelshift: {
+        valType: 'integer',
+        dflt: 0,
+        editType: 'ticks',
+        description: [
+            'Shifts the tick labels by the specified number of pixels in parallel to the axis.',
+            'Positive values move the labels in the positive direction of the axis.'
+        ].join(' ')
+    },
+    ticklabelstandoff: {
+        valType: 'integer',
+        dflt: 0,
```

Comment:
I think we could/should set `min: 0` for this one as negative values are not very useful and they could be confusing e.g. outside labels could move inside, etc.

Reply:
Wait. I am still not sure about restricting this option.
It looks like people may want to have that as illustrated in the #1673 graph.
If that's the case then I would suggest that we mention in the description that negative `standoff` values could move outside ticks inside the subplot and vice versa.

Reply:
I added the description. Actually, the negative standoff is the main use case for us because that makes it possible to make the chart a little bit smaller.